qr-code/src/main/java/com/adyen/checkout/qrcode/internal/ui/model/QRCodePaymentMethodConfig.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by onurk on 7/3/2023.
+ */
+
+package com.adyen.checkout.qrcode.internal.ui.model
+
+import androidx.annotation.StringRes
+import com.adyen.checkout.components.core.internal.util.PaymentMethodTypes
+import com.adyen.checkout.qrcode.R
+import com.adyen.checkout.qrcode.internal.ui.QrCodeComponentViewType
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+internal enum class QRCodePaymentMethodConfig(
+    private val paymentMethodType: String,
+    val maxPollingDurationMillis: Long,
+    val viewType: QrCodeComponentViewType,
+    @StringRes val messageTextResource: Int
+) {
+    DUIT_NOW(
+        paymentMethodType = PaymentMethodTypes.DUIT_NOW,
+        maxPollingDurationMillis = DUIT_NOW_MAX_POLLING_DURATION,
+        viewType = QrCodeComponentViewType.FULL_QR_CODE,
+        messageTextResource = R.string.checkout_qr_code_duit_now
+    ),
+    PAY_NOW(
+        paymentMethodType = PaymentMethodTypes.PAY_NOW,
+        maxPollingDurationMillis = PAY_NOW_MAX_POLLING_DURATION,
+        viewType = QrCodeComponentViewType.FULL_QR_CODE,
+        messageTextResource = R.string.checkout_qr_code_pay_now
+    ),
+    UPI_QR(
+        paymentMethodType = PaymentMethodTypes.UPI_QR,
+        maxPollingDurationMillis = UPI_MAX_POLLING_DURATION,
+        viewType = QrCodeComponentViewType.FULL_QR_CODE,
+        messageTextResource = R.string.checkout_qr_code_upi
+    );
+
+    companion object {
+        fun getByPaymentMethodTye(paymentMethodType: String): QRCodePaymentMethodConfig? {
+            return values().firstOrNull { it.paymentMethodType == paymentMethodType }
+        }
+    }
+}
+
+private val DUIT_NOW_MAX_POLLING_DURATION = 90.seconds.inWholeMilliseconds
```

Comment:
These should be in the companion object for consistency.

Reply:
I can't do this because of following explanation
https://stackoverflow.com/questions/57560580/kotlin-enum-constructor-argument-must-be-initialized-with-companion-object-con

Reply:
Do we need these variables anyway? If we only use them to initialize the enum how about we directly use `maxPollingDurationMillis = 90.seconds.inWholeMilliseconds,`?

Reply:
I thought it but I was not sure about whether they are magic number.

Reply:
What @jreij suggested sounds good to me too. I think it's fine to use it directly in the definition.